projects/ngx-d3timeline/src/lib/axis/resources-axis/resource-axis-tick-renderer.ts

```diff
@@ -1,8 +1,16 @@
 import { BandScale } from '../../scale-types';
 import { TickRenderer } from '../tick-renderer';
+import { Orientation } from '../../orientation';
+import { OptionsService } from '../../options.service';
+import { TimelineView } from '../../view/timeline-view';
 
 export class ResourceAxisTickRenderer implements TickRenderer {
-  constructor(private scale: BandScale) {}
+  constructor(
+    private scale: BandScale,
```

Comment:
only `optionsService` should be part of the constructor, the other arguments should be passed into the functions. This will allow the renderer to be injected into the axis service

Reply:
I didn't like the number of parameters here purely based on how many there were.

Passing them as arguments to all the methods will cause a lot of repetition and verbosity. 

If we want to inject the renderers I think there are more wide scale changes required in particular to the AxisService to think about.

I had a plan for addressing the number of params but thinking it through it prob won't work.

What benefits do you see from injecting the renderers?

Reply:
I guess the benefit is looser coupling between the renderers and the axis service. One consequence of this is the ability to test the renderer and axis service in isolation. Also by passing all the other parameters into the constructor we're adding state to the renderer. The functions in the renderer are no longer pure.

Reply:
OK, I'll give that a go in #107 to prevent bloating this pull request